File: internal/adapters/driver/lambda/check_payment_status.go
package driver_lambda

import (
	"context"

	"github.com/aws/aws-lambda-go/events"
	"github.com/fiap/ms-payment/internal/core/application/usecases"
	"github.com/fiap/ms-payment/pkg/utils"
)

type CheckPaymentStatusHandler struct {
	useCase *usecases.CheckPaymentStatusUseCase
}

func NewCheckPaymentStatusHandler(useCase *usecases.CheckPaymentStatusUseCase) APIGatewayHandler {
	return &CheckPaymentStatusHandler{useCase: useCase}
}

func (h *CheckPaymentStatusHandler) Handle(ctx context.Context, event events.APIGatewayProxyRequest) (events.APIGatewayProxyResponse, error) {
	externalId := event.PathParameters["external_id"]
	if externalId == "" {
		return utils.HandleHTTPError(utils.HTTPBadRequest("missing external_id in path parameters")), nil
	}

	res, err := h.useCase.Execute(ctx, externalId)
	if err != nil {
		return utils.HandleHTTPError(err), nil
	}

	return utils.HTTPSuccess(res), nil
}

File: src/adapters/driver/lambda/check_payment_status/check_payment_status.ts
import { APIGatewayProxyEvent, APIGatewayProxyResult } from "aws-lambda"

import { CheckPaymentStatusContainerFactory } from "@di/check_payment_status"
import { HTTPBadRequest, HTTPError, HTTPSuccessResponse } from "@utils/http"
import { getRegion, getStage } from "@utils/env"

export async function handler(event: APIGatewayProxyEvent): Promise<APIGatewayProxyResult> {
	try {
		const externalId = event.pathParameters?.external_id
		if (!externalId) {
			throw new HTTPBadRequest("missing external_id in path parameters")
		}

		const stage = getStage()
		const region = getRegion()

		const container = new CheckPaymentStatusContainerFactory(region, stage)

		const res = await container.usecase.execute(externalId)

		return new HTTPSuccessResponse(res).toLambdaResponse()
	} catch (error) {
		if (error instanceof HTTPError) {
			return error.toLambdaResponse()
		}

		console.error("Unexpected error:", error)

		const genericError = new HTTPError("Internal Server Error", 500)
		return genericError.toLambdaResponse()
	}
}
